src/components/smart_objects/src/smart_object.cc

```diff
@@ -723,16 +723,28 @@ void SmartObject::duplicate(const SmartObject& OtherObject) {
 void SmartObject::cleanup_data() {
```

Comment:
@BrandonHe It is very strange to see modification of logic request controller and smart object in one pull request. Please describe what for do you need this changes.